src/main.rs

```diff
@@ -104,6 +107,9 @@ async fn main() -> Result<()> {
             .layer(CatchPanicLayer::custom(runtime::catch_panic))
             // Mark headers as sensitive on both requests and responses.
             .layer(SetSensitiveHeadersLayer::new([header::AUTHORIZATION]))
+            // here is openAPI with UAT:
+            // TODO: link docs on merge
+            // Here is where I add new functionality, right?
             .merge(SwaggerUi::new("/swagger-ui").url("/api-doc/openapi.json", ApiDoc::openapi()));
```

Comment:
[router.merge](https://docs.rs/axum/latest/axum/struct.Router.html#method.merge)
Merge the paths and fallbacks of two routers into a single [Router](https://docs.rs/axum/latest/axum/struct.Router.html).

Router is the type for composing [handlers](https://docs.rs/axum/latest/axum/handler/index.html) and services.

https://blog.logrocket.com/rust-axum-error-handling/

Reply:
SwaggerUI is imported at L29;
ApiDoc at L31

Reply:
ah, ok so if I add a new [layer](https://docs.rs/axum/latest/axum/struct.Router.html#method.layer) it should fit with everything else, right?

Reply:
This can be used to add additional processing to a request for a group of routes.

Reply:
https://docs.rs/axum/latest/axum/index.html#routing

Reply:
In axum a “handler” is an async function that accepts zero or more [“extractors”](https://docs.rs/axum/latest/axum/extract/index.html) as arguments and returns something that can be converted [into a response](https://docs.rs/axum/latest/axum/response/index.html).

Handlers are where your application logic lives and axum applications are built by routing between handlers.

Reply:
https://github.com/ulid/spec

Reply:
https://blog.logrocket.com/rust-axum-error-handling/